docs/HowToGuides/GettingStarted.md

```diff
@@ -365,78 +365,65 @@ whenever the heading is modified.
-  * In the _Info_ pane, select the _Executable_ built by the Ninja target from
-    the appropriate `bin` directory (e.g. `build/Ninja-*/swift-macosx-*/bin/swift-frontend`).
-  * In the _Arguments_ pane, add the command line arguments that you want to
-    pass to the executable on launch (e.g. `path/to/file.swift -typecheck` for
+  * _Arguments_ to a Ninja target (e.g. `bin/swift-frontend`, the compiler)
+  * _Directory_ to the absolute path of the `build/Ninja-*/swift-macosx-*`
+    directory
+* Create a scheme in the workspace, making sure to select the target you just
+  created. Be *extra* careful not to choose a target from the generated Xcode
+  project you added to the workspace.
+* Spot-check your target in the settings for the _Build_ scheme action.
+* If the target is executable, adjust the settings for the _Run_ scheme action:
+  * In the _Info_ pane, select the _Executable_ produced by the Ninja target
+    from `build/Ninja-*/swift-macosx-*/bin` (e.g. `swift-frontend`).
+  * In the _Arguments_ pane, add command line arguments that you want to pass to
+    the executable on launch (e.g. `path/to/file.swift -typecheck` for
     `bin/swift-frontend`).
-  * You can optionally set the working directory for debugging in the
-    _Options_ pane.
-* Configure as many more target-scheme pairs as you need.
-
-Now you are all set! You can build, run and debug as with a native Xcode
-project. If an `update-checkout` routine or a structural change—such as when
-source files are added or deleted—happens to impact your editing experience,
-simply regenerate the Xcode projects.
-
-> **Note**  
-> * For debugging to *fully* work for a given component—say, the compiler—the
-    `build-script` invocation for the Ninja build must be arranged to
-    [build a debug variant of that component](#debugging-issues).
-> * Xcode's indexing can occasionally start slipping after switching to and back
-    from a distant branch, resulting in a noticeable slowdown. To sort things
-    out, close the workspace and delete the _Index_ directory from the
-    workspace's derived data before reopening.
+  * Optionally set a custom working directory in the _Options_ pane.
+* Follow the previous steps to create more targets and schemes per your line
+  of work.
+
+#### Regenerating the Xcode project
+
+The structure of the generated Xcode project is distinct from the underlying
+organization of the files on disk, and does not adapt to changes in the file
+system, such as file/directory additions/deletions/renames. Over the course of
+multiple `update-checkout` rounds, the resulting divergence is likely to begin
+affecting your editing experience. To fix this, simply regenerate the project
+by running the invocation from the first step.
+
+#### Troubleshooting editing issues in Xcode
+
+* If a syntax highlighting or code action issue does not resolve itself after
+  regenerating the Xcode project, try *Product > Analyze*.
```

Comment:
`Product > Analyze` only works if there is an active scheme. Is there a default one we should be using for the `swift.xcodeproj`?

Reply:
The idea is to use the scheme that covers the affected area. There is a Ninja target you can hook up that builds everything, but I’ve never tried running static analysis against it, nor do I even use it. Are you experiencing issues with source tooling?

Reply:
Is the source tooling setup of the external build tool?

Reply:
> Is the source tooling setup of the external build tool?

Sorry, I’m not sure I understood that right. By source tooling I mean Xcode’s syntax highlighting and features like jump to definition.

Reply:
so that is what you meant by source tooling. Yes, I am having issues with that

Reply:
Which area/files?

Reply:
Because the `swiftASTGen` scheme was built, I was able to run `Product -> Analyze` to fix the source tooling.

Reply:
Ah, there you go. Very weird that building `swiftASTGen` was not enough to wake Xcode though.

Reply:
regarding the `target` and `scheme`

I see that there now exists a `swift-frontend target`. But, the docs say to NOT create a `scheme` based off of the autogenerated targets. So now I should create a new `scheme` for `swift-frontend`?

Reply:
> I should NOT do what I am doing below

Yes. Instead, you should follow the steps to manually creating an external build system target and then creating a scheme with it.

Reply:
i checked the path of the targets added and they were located in the `swift-macosx-*/`,  this is wrong.

We want to use the executables in the `build/Ninja-*/swift-macosx-*` , so I have to manually create the external build system. 

Finally, this build WILL contribute to the build artifacts in `swift-macosx-*/` as you stated above.

Reply:
![Screen Shot 2023-09-16 at 5 59 20 PM](https://github.com/apple/swift/assets/20051028/f61b7b00-caa9-4d02-bd4d-8b83af8e15ee)

weird how the `swift-frontend` build succeeds but still getting a fatal error 🤔

Reply:
> i checked the path of the targets added and they were located in the `swift-macosx-*/`,  this is wrong.

Could you elaborate on what exactly is wrong?

> Finally, this build WILL contribute to the build artifacts in `swift-macosx-*/` opposite to what you stated the `ASTGen` scheme does.  

Right, the targets you create manually according to the guide will contribute to the Ninja build artifacts.

Reply:
> Could you elaborate on what exactly is wrong?

Nothing is wrong, but I am just rubber ducking/thinking out loud why I can't use the autogenerated targets.

Reply:
> weird how the swift-frontend build succeeds but still getting a fatal error

Are you passed arguments to the executable? Invoking `swift-frontend` without arguments used to launch the now obsoleted integrated REPL, hence the error.

Reply:
Is there a `--help` command to see arguments that are able to be passed?

Reply:
For binaries and scripts here it's always either `-h`, `-help` or `--help`.

> I just don't understand why we need to pass a single file to the executable. Where does this come from and how do we know what to pass? 

Because most of the time a single file is enough for preliminary testing. The baseline workflow for fixing bugs or implementing language features goes something like this:
1. Prepare a file with a test case
1. Make changes
1. Build whatever needs to be built
1. Run the relevant binary against this file to debug/test your changes. If anything's wrong, go to step 2
1. Run relevant test suite areas. If anything regresses, go to step 2
1. Add regression tests and run them. Unexpected failure? Go to step 2
1. Tidy up changes & commit history

If a task is not expected to impact language behavior, the workflow would normally be limited to steps 2-3, 5 and 7.

Reply:
is there a good way of recognizing impact language rules/behavior? For an issue like #68350 , my gut is telling me I only need to do steps 2-3, 5 and 7, but I would like to understand fully why.

Reply:
also are there any notes on the doc for adding the `swift/test` directory for updating/creating tests?

Reply:
> is there a good way of recognizing impact language rules/behavior? For an issue like #68350 , my gut is telling me I only need to do steps 2-3, 5 and 7, but I would like to understand fully why.

I believe recognizing whether a task is *expected* to have an observable effect on the end product is mostly a matter of understanding the goals of that task. Your intuition isn’t letting you down. Where is the doubt stemming from?

Reply:
it is fairly easy to changes to pieces of code without having an actual understanding of what is going on. I would like to understand what the changes I made actually do and how it affects other pieces. This is a very large codebase and I do want to take the time to understand so I can contribute more in the future.

Reply:
This looks like a great discussion. Would you be willing to move this discussion to forums.swift.org (i.e. Development category) instead? It would be more visible to participate in and find in the future.

Reply:
Great! @AnthonyLatsis could you help with that?

Reply:
> Is running this command supposed to cause these errors?
 
No, but it should be safe to ignore them. Did the logs show why `python_lint.swift` failed?

Reply:
missing installed libraries `flake8` 🤦🏽‍♂️

Reply:
> Great! @AnthonyLatsis could you help with that?

Thanks, Pavel, I got this. @saehejkang Let me know about your username once you sign up.

Reply:
Thank you!

Reply:
> Let me know about your username once you sign up.

same username as my github

Reply:
https://forums.swift.org/t/an-account-on-getting-started-with-contributing-code/67372